src/main/java/com/o19s/es/ltr/feature/store/StoredFeatureNormalizers.java

```diff
@@ -0,0 +1,197 @@
+        this.featureNormalizers = new HashMap<>();
+        int numFeatureNorms = input.readInt();
+        for (int i = numFeatureNorms; i > 0; i--) {
+            FeatureNormDefinition norm = this.createFromStreamInput(input);
+            this.featureNormalizers.put(norm.featureName(), norm);
+        }
+    }
+
+    public Normalizer getNormalizer(String featureName) {
+        return this.featureNormalizers.get(featureName).createFeatureNorm();
+    }
+
+    public static final String TYPE = "feature_normalizers";
+
+
+    public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
+        builder.startObject(); // begin feature norms
+        for (Map.Entry<String, FeatureNormDefinition> ftrNormDefEntry: featureNormalizers.entrySet()) {
+            builder.field(ftrNormDefEntry.getKey());
+            ftrNormDefEntry.getValue().toXContent(builder, params);
+        }
+        builder.endObject(); // feature norms
+        return builder;
+    }
+
+    public void addFeatureNormalizers(List<FeatureNormDefinition> ftrNormDefs) {
+        this.featureNormalizers = new HashMap<>();
+        for (FeatureNormDefinition ftrNorm: ftrNormDefs) {
+            this.featureNormalizers.put(ftrNorm.featureName(), ftrNorm);
+        }
+    }
+
+    public FeatureNormalizerSet compile(FeatureSet featureSet) {
+        if (featureNormalizers.size() == 0) {
+            return new NoOpFeatureNormalizerSet();
+        }
+        List<Normalizer> ftrNorms = new ArrayList<>(featureSet.size());
+
+        for (int i = 0; i < featureSet.size(); i++) {
+            ftrNorms.add(new NoOpNormalizer());
+        }
+
+        for (Map.Entry<String, FeatureNormDefinition> ftrNormDefEntry: featureNormalizers.entrySet()) {
+            String featureName = ftrNormDefEntry.getValue().featureName();
+            Normalizer ftrNorm = ftrNormDefEntry.getValue().createFeatureNorm();
+
+            if (!featureSet.hasFeature(featureName)) {
+                throw new ElasticsearchException("Feature " + featureName +
+                                                 " not found in feature set " + featureSet.name());
+            }
+
+            int ord = featureSet.featureOrdinal(featureName);
+            ftrNorms.set(ord, ftrNorm);
+        }
+        return new CompiledFeatureNormalizerSet(ftrNorms);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof StoredFeatureNormalizers)) return false;
+        StoredFeatureNormalizers that = (StoredFeatureNormalizers) o;
+
+        return that.featureNormalizers.equals(this.featureNormalizers);
+    }
+
+    @Override
+    public int hashCode() {
+        return this.featureNormalizers.hashCode();
+    }
+
+
+    private  FeatureNormDefinition createFromStreamInput(StreamInput input) throws IOException {
+        Type normType = Type.readFromStream(input);
```

Comment:
you should perhaps use the registry:
In the plugin class in getNamedWritables you could add the entries:
```java
new Entry(FeatureNormDefinition.class, MinMaxFeatureNormDefinition.TYPE, MinMaxFeatureNormDefinition::new)
```

This way you just have to call:
```List<FeatureNormDefinition> featureNormalizers = input.readNamedWritableList(FeatureNormDefinition.class)```
in the constructor of this class.